grammar.js

```diff
@@ -30,42 +32,67 @@ module.exports = grammar({
   name: "git_commit",
 
   extras: ($) => [WHITE_SPACE],
-
   rules: {
     source: ($) =>
       seq(
-        optional(choice($.comment, $.subject)),
-        optional(seq(NEWLINE, repeat($._body_line))),
+        optional(choice($.subject, $.comment)),
+        repeat($._body_line),
+        repeat($._trailer),
+        optional(repeat1($._newline)),
         optional(
           seq(alias(SCISSORS, $.scissors), optional(alias($._rest, $.message)))
         )
       ),
-
-    _body_line: ($) =>
-      choice(seq($.message, NEWLINE), seq($.comment, NEWLINE), NEWLINE),
-
+    _newline: ($) => NEWLINE,
+    /**
+     * The subject of the commit message: the first line.
+     */
     subject: ($) =>
       seq(
-        token(prec(PREC.SUBJECT_FIRST_CHAR, /[^#\r\n]/)),
-        optional(repeat(/[^\r\n]+/))
+        prec(PREC.SUBJECT_FIRST_CHAR, token.immediate(/[^#\r\n]/)),
+        repeat(ANYTHING)
       ),
 
+    _body_line: ($) =>
+      prec.right(1, seq($._newline, optional(choice($.message, $.comment)))),
+    _trailer: ($) => seq($._newline, $.trailer),
+    /**
+     * Non-comment body
+     */
     message: ($) =>
       seq(
-        repeat(WHITE_SPACE),
-        choice($.user, /[^\s#]+/, /\s+/),
-        optional(repeat(choice($.user, $.item, $._word)))
+        // beginning at the start of the line
+        repeat(WHITE_SPACE), // optional leading whitespace
+        choice($.user, $.commit, $._word),
+        repeat(choice($.user, $.item, $.commit, $._word))
       ),
 
-    comment: ($) => seq("#", optional($._comment_body)),
-
-    _comment_body: ($) =>
-      choice(
-        alias($._rebase_summary, $.summary),
-        $.summary,
-        $._branch_declaration,
-        // fallback to regular comment words if the words are nonsense
-        repeat1($._word)
+    /**
+     * Trailers are "lines that look similar to RFC 822 e-mail headers at the
+     * end of the otherwise free-form part of a commit message." For more details
+     * on the format, see https://git-scm.com/docs/git-interpret-trailers
+     */
+    trailer: ($) =>
+      seq(
+        field("key", $._word),
+        /[:=]/,
```

Comment:
```suggestion
        field("separator", choice(":", "=")),
```

This will make it easy to theme the separator with syntax highlighting queries later, either:

```scm
(trailer separator: _ @punctuation.delimiter)
```

or

```scm
[":" "="] @punctuation.delimiter
```